src/figuresClasses.ts

```diff
@@ -1,19 +1,110 @@
-export interface Figure {
+export enum Color {
+  red,
+  green,
+  blue,
+}
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea: Function;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = Shape.Triangle;
+
+  longestSide = 0;
+
+  sides: [number, number, number];
+
+  sumOfSides: number;
+
+  constructor(
+    public color: Color,
+    ...args: [number, number, number]
+  ) {
+    this.sides = [...args];
+
+    this.sides.forEach((side) => {
+      if (side > this.longestSide) {
+        this.longestSide = side;
+      }
+
+      if (side <= 0) {
+        throw new Error('One of the sides is too short');
+      }
+    });
 
+    this.sumOfSides = [...args].reduce((partialSum, a) => partialSum + a, 0);
+
+    if (this.longestSide >= (this.sumOfSides - this.longestSide)) {
+      throw new Error('Triangle proportion is incorrect');
+    }
```

Comment:
Will be better to create separate function, that will accept initial parameters, check it and return the corresponding error if not
And you can call this one general function in each class